Yumemi-ios-training/Repository/WeatherRepository.swift

```diff
@@ -15,13 +15,19 @@ protocol WeatherRepositoryProtocol: AnyObject {
 }
 
 final class WeatherRepository: WeatherRepositoryProtocol {
+    private let weatherClient: WeatherClientProtocol
+    
+    init(weatherClient: WeatherClientProtocol) {
+        self.weatherClient = weatherClient
+    }
+    
     func fetchWeather() -> Weather {
-        return WeatherClient.fetchWeather()
+        return self.weatherClient.fetchWeather()
```

Comment:
[FYI]
```suggestion
        weatherClient.fetchWeather()
```

1 行の return は省略できますのでご参考までに。